ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch_topology.go

```diff
@@ -357,6 +361,10 @@ func elasticsearchTopologyAutoscalingPayload(ctx context.Context, autoObj attr.V
 		}
 	}
 
+	if !autoscale.TierAutoscale.IsNull() && !autoscale.TierAutoscale.IsUnknown() {
+		payload.AutoscalingTierOverride = autoscale.TierAutoscale.ValueBoolPointer()
```

Comment:
IMO it could be that `payload.AutoscalingTierOverride` gets `false` but lines 195-198 allow only `true` value to be assigned to `a.TierAutoscale`. Can it lead to an error in the provider when the read value doesn't correspond to the set one? (I understand that from the backend perspective `nil` and `false` are semantically identical as we discussed).

Reply:
Indeed `nil` and `false` are equivalent from the backend perspective, but that's a good catch 👍 

https://github.com/elastic/terraform-provider-ec/pull/761/commits/d9c161769e7ba5f37ee041d85ba2d1ffca0c49e0 addressed that.